File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
using namespace std;


class arrStack {
private:
	int* arr;
	int capacity;
	int t;

public:
	arrStack(int capacity) {
		this->capacity = capacity;
		this->t = -1;
		this->arr = new int[capacity];
	}

	int size() {
		return t + 1;
	}

	bool empty() {
		return t == -1;
	}

	int top() {
		if (empty())
			return -1;
		else
			return arr[t];
	}

	void push(int e)
	{
		if (size() == capacity)
			return;
		else
		{
			t = t + 1;
			arr[t] = e;
		}
	}

	void pop() {
		if (empty())
			cout << "-1" << endl;
		else
		{
			cout << arr[t] << endl;
			t = t - 1;
		}
	}

};





int main() {
	int N;
	cin >> N;

	arrStack arrStack(N);

	for (int i = 0; i < N; i++)
	{
		string op;
		cin >> op;

		if (op == "push")
		{
			int val;
			cin >> val;
			arrStack.push(val);
		}

		if (op == "top")
			cout << arrStack.top() << endl;

		if (op == "size")
			cout << arrStack.size() << endl;

		if (op == "pop")
			arrStack.pop();
		if (op == "empty")
		{
			if (arrStack.empty())
				cout << "1" << endl;
			else
				cout << "0" << endl;
		}


	}

}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;
import java.util.stream.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.valueOf(br.readLine());
        Deque<Integer> dq = new ArrayDeque<>();

        for(int i = 0 ; i < N ; i++) {
            String[] op = br.readLine().split(" ");
            String cur = op[0];

            if(cur.equals("push")){
                dq.offerLast(Integer.valueOf(op[1]));
            }else if(cur.equals("pop")){
                if(dq.isEmpty()) System.out.println(-1);
                else System.out.println(dq.pollLast());
            }else if(cur.equals("size")){
                System.out.println(dq.size());
            }else if(cur.equals("empty")){
                if(dq.isEmpty()) System.out.println(1);
                else System.out.println(0);
            }else if(cur.equals("top")){
                if(dq.isEmpty()) System.out.println(-1);
                else System.out.println(dq.peekLast());
            }

        }


    }
}
